src/compiler/compile/css/Stylesheet.ts

```diff
@@ -376,6 +376,35 @@ export default class Stylesheet {
 		}
 	}
 
+	recompute_stylesheet_hash(component: Component) {
+		const collectIf = (test_function) => (acc, child) => {
+			if (child instanceof Atrule) {
+				child.children.forEach(rule => {
+					if (test_function(rule)) {
+						acc.push(rule);
+					}
+				});
+			} else {
+				if (test_function(child)) {
+					acc.push(child);
+				}
+			}
+			return acc;
+		};
+
+		const used_rules = this.children.reduce(collectIf(rule => rule.is_used(component.compile_options.dev)), []);
+		const unused_rules = this.children.reduce(collectIf(rule => !rule.is_used(component.compile_options.dev)), []);
+		if (unused_rules.length > 0) {
+			const new_hash = hash(used_rules.reduce((acc, rule: Rule) => {
+				for (let i = rule.node.start; i < rule.node.end; i++) {
+					acc += component.source.charAt(i);
+				}
+				return acc;
+			}, ''));
+			this.id = `svelte-${new_hash}`;
```

Comment:
to me, reassigning the stylesheet `id` here seems clunky even though it does fix the reported bug.  i would've preferred to set the `id` accounting for the unused selectors [in the `Stylesheet` constructor](https://github.com/sveltejs/svelte/blob/master/src/compiler/compile/css/Stylesheet.ts#L301) but, at this stage, [`Rule.is_used()`](https://github.com/sveltejs/svelte/blob/master/src/compiler/compile/css/Stylesheet.ts#L58-L62) only returns `false`.  it isn't until [after the html `Fragment` is created](https://github.com/sveltejs/svelte/blob/master/src/compiler/compile/Component.ts#L168) that the dom nodes are paired with the css selectors and [`Rule.is_used()`](https://github.com/sveltejs/svelte/blob/master/src/compiler/compile/css/Stylesheet.ts#L58-L62) can accurately report which are unused.  any suggestions on how to improve this flow would be appreciated.